2024-05-29 16:06:54,897: asyncio: INFO: [MLTrader] Executing the initialize lifecycle method
2024-05-29 16:06:55,373: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)
Traceback (most recent call last):
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca/trading/stream.py", line 168, in _run_forever
    await self._start_ws()
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca/trading/stream.py", line 128, in _start_ws
    await self._connect()
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca/trading/stream.py", line 59, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1098, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1131, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 577, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 559, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)
2024-05-29 16:06:55,453: alpaca.trading.stream: INFO: starting trading websocket connection
2024-05-29 16:06:55,597: root: INFO: Current backtesting datetime 2023-12-26 08:30:00-05:00
2024-05-29 16:06:55,598: asyncio: INFO: [MLTrader] Executing the before_market_opens lifecycle method
2024-05-29 16:06:55,598: root: INFO: Current backtesting datetime 2023-12-26 09:30:00-05:00
2024-05-29 16:06:55,599: asyncio: INFO: [MLTrader] Executing the before_starting_trading lifecycle method
2024-05-29 16:06:55,599: asyncio: INFO: [MLTrader] [34mExecuting the on_trading_iteration lifecycle method at 2024-05-29 16:06:55[0m
2024-05-29 16:06:55,599: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-05-29 16:06:55,933: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)
Traceback (most recent call last):
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca/trading/stream.py", line 168, in _run_forever
    await self._start_ws()
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca/trading/stream.py", line 128, in _start_ws
    await self._connect()
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca/trading/stream.py", line 59, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1098, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1131, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 577, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 559, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)
2024-05-29 16:06:55,944: alpaca.trading.stream: INFO: starting trading websocket connection
2024-05-29 16:06:56,430: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)
Traceback (most recent call last):
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca/trading/stream.py", line 168, in _run_forever
    await self._start_ws()
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca/trading/stream.py", line 128, in _start_ws
    await self._connect()
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca/trading/stream.py", line 59, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1098, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1131, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 577, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 559, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)
2024-05-29 16:06:56,442: alpaca.trading.stream: INFO: starting trading websocket connection
2024-05-29 16:06:56,855: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)
Traceback (most recent call last):
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca/trading/stream.py", line 168, in _run_forever
    await self._start_ws()
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca/trading/stream.py", line 128, in _start_ws
    await self._connect()
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca/trading/stream.py", line 59, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1098, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1131, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 577, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 559, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)
2024-05-29 16:06:56,866: alpaca.trading.stream: INFO: starting trading websocket connection
2024-05-29 16:06:57,252: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)
Traceback (most recent call last):
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca/trading/stream.py", line 168, in _run_forever
    await self._start_ws()
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca/trading/stream.py", line 128, in _start_ws
    await self._connect()
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca/trading/stream.py", line 59, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1098, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1131, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 577, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 559, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)
2024-05-29 16:06:57,264: alpaca.trading.stream: INFO: starting trading websocket connection
2024-05-29 16:06:57,378: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-05-29 16:06:57,653: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)
Traceback (most recent call last):
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca/trading/stream.py", line 168, in _run_forever
    await self._start_ws()
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca/trading/stream.py", line 128, in _start_ws
    await self._connect()
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca/trading/stream.py", line 59, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1098, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1131, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 577, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 559, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)
2024-05-29 16:06:57,665: alpaca.trading.stream: INFO: starting trading websocket connection
2024-05-29 16:06:57,989: asyncio: ERROR: [MLTrader] 403 Client Error: Forbidden for url: https://data.alpaca.markets/v1beta1/news?start=2023-12-23&end=2023-12-26&limit=10&sort=desc&include_content=False&exclude_contentless=False&symbols=SPY
2024-05-29 16:06:57,992: asyncio: ERROR: [MLTrader] Traceback (most recent call last):
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/lumibot/strategies/strategy_executor.py", line 923, in run
    self._run_trading_session()
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/lumibot/strategies/strategy_executor.py", line 871, in _run_trading_session
    self._on_trading_iteration()
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/lumibot/strategies/strategy_executor.py", line 262, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/lumibot/strategies/strategy_executor.py", line 286, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/lumibot/strategies/strategy_executor.py", line 397, in _on_trading_iteration
    raise e
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/lumibot/strategies/strategy_executor.py", line 369, in _on_trading_iteration
    on_trading_iteration()
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/lumibot/tools/decorators.py", line 62, in func_output
    frame, result = call_function_get_frame(func_input, *args, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/lumibot/tools/decorators.py", line 30, in call_function_get_frame
    result = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/trading_bot.py", line 66, in on_trading_iteration
    probability, sentiment = self.get_sentiment_of_news()
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/trading_bot.py", line 51, in get_sentiment_of_news
    news = self.api.get_news(symbol = self.symbol,
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca_trade_api/rest.py", line 988, in get_news
    news = list(self.get_news_iter(symbol=symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca_trade_api/rest.py", line 972, in get_news_iter
    for n in news:
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca_trade_api/rest.py", line 81, in raise_api_error
    raise http_error from None
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca_trade_api/rest.py", line 243, in _one_request
    resp.raise_for_status()
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/requests/models.py", line 1024, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: https://data.alpaca.markets/v1beta1/news?start=2023-12-23&end=2023-12-26&limit=10&sort=desc&include_content=False&exclude_contentless=False&symbols=SPY

2024-05-29 16:06:57,992: asyncio: INFO: [MLTrader] Executing the on_bot_crash event method
2024-05-29 16:06:57,995: root: INFO: Backtesting finished
2024-05-29 16:06:58,055: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)
Traceback (most recent call last):
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca/trading/stream.py", line 168, in _run_forever
    await self._start_ws()
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca/trading/stream.py", line 128, in _start_ws
    await self._connect()
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca/trading/stream.py", line 59, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1098, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1131, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 577, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 559, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)
2024-05-29 16:06:58,067: alpaca.trading.stream: INFO: starting trading websocket connection
2024-05-29 16:06:58,463: alpaca.trading.stream: ERROR: error during websocket communication: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)
Traceback (most recent call last):
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca/trading/stream.py", line 168, in _run_forever
    await self._start_ws()
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca/trading/stream.py", line 128, in _start_ws
    await self._connect()
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/alpaca/trading/stream.py", line 59, in _connect
    self._ws = await websockets.connect(self._endpoint, **self._websocket_params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/websockets/legacy/client.py", line 659, in __await_impl_timeout__
    return await asyncio.wait_for(self.__await_impl__(), self.open_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/Users/sreekargudipati/Coding Projects/AutoTrader/myenv/lib/python3.11/site-packages/websockets/legacy/client.py", line 663, in __await_impl__
    _transport, _protocol = await self._create_connection()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1098, in create_connection
    transport, protocol = await self._create_connection_transport(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 1131, in _create_connection_transport
    await waiter
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 577, in _on_handshake_complete
    raise handshake_exc
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/sslproto.py", line 559, in _do_handshake
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py", line 979, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)
2024-05-29 16:06:58,474: alpaca.trading.stream: INFO: starting trading websocket connection
2024-05-29 16:06:58,771: asyncio: WARNING: [MLTrader] Cannot plot returns because the strategy returns are missing
2024-05-29 16:06:58,813: asyncio: WARNING: [MLTrader] Cannot create a tearsheet because the strategy returns are missing
2024-05-29 16:06:58,813: backtest_stats: INFO: Backtest took 0:00:03.917613 for a speed of 0.000
